src/Simulation/Simulation.jl

```diff
@@ -176,112 +176,85 @@ end
 # Functions
 
 function Grid(  sim::Simulation{T, Cylindrical};
-                init_grid_size::Union{Missing, NTuple{3, Int}} = missing,
-                init_grid_spacing::Union{Missing, Tuple{<:Real,<:Real,<:Real}} = missing,
                 for_weighting_potential::Bool = false,
-                min_n_ticks::Int = 10,
-                max_tick_distance::Real = 1e-2,
+                max_tick_distance::Union{Missing, length_unit, Tuple{length_unit, angle_unit, length_unit}} = missing,
+                max_distance_ratio::Real = 5,
                 full_2π::Bool = false)::CylindricalGrid{T} where {T}
-    if ismissing(init_grid_size)
-        world_diffs = [(getproperty.(sim.world.intervals, :right) .- getproperty.(sim.world.intervals, :left))...]
-        world_diffs[2] = world_diffs[2] * 0.3 * sim.world.intervals[1].right # in radiance
-        inds::Vector{Int} = sortperm([world_diffs...])
-        ratio::T = min_n_ticks * if world_diffs[inds[1]] > 0
-            inv(world_diffs[inds[1]])
-        elseif world_diffs[inds[2]] > 0
-            inv(world_diffs[inds[2]])
-        elseif world_diffs[inds[3]] > 0
-            inv(world_diffs[inds[3]])
-        else
-            error("This should not happen... World has no dimension")
-        end
-        init_grid_size_1::Int = convert(Int, round(ratio * world_diffs[1], RoundUp))
-        init_grid_size_2::Int = convert(Int, round(ratio * world_diffs[2], RoundUp))
-        init_grid_size_3::Int = convert(Int, round(ratio * world_diffs[3], RoundUp))
-        init_grid_size::NTuple{3, Int} = NTuple{3, T}( [init_grid_size_1, init_grid_size_2, init_grid_size_3] )
-    end
-
-    init_grid_spacing, use_spacing::Bool = if !ismissing(init_grid_spacing)
-        T.(init_grid_spacing), true
-    else
-        missing, false
-    end
-    
-    samples::Vector{CylindricalPoint{T}} = sample(sim.detector, Cylindrical)
-   
+    detector = sim.detector
+    world = sim.world 
+                
+    samples::Vector{CylindricalPoint{T}} = sample(detector, Cylindrical)
     important_r_points::Vector{T} = map(p -> p.r, samples)
     important_φ_points::Vector{T} = map(p -> p.φ, samples)
     important_z_points::Vector{T} = map(p -> p.z, samples)
 
-    push!(important_r_points, sim.world.intervals[1].left)
-    push!(important_r_points, sim.world.intervals[1].right)
+    world_Δz = world.intervals[3].right - world.intervals[3].left
+    world_Δφ = world.intervals[2].right - world.intervals[2].left
+    world_Δr = world.intervals[1].right - world.intervals[1].left
+    world_r_mid = (world.intervals[1].right + world.intervals[1].left)/2
+    
+    max_distance_z = T(world_Δz / 4)
+    max_distance_φ = T(world_Δφ / 4)
+    max_distance_r = T(world_Δr / 4)
+    if !ismissing(max_tick_distance)
+        if max_tick_distance isa Quantity{<:Real, Unitful.𝐋}
```

Comment:
You defined `Quantity{<:Real, Unitful.𝐋}` as `length_unit` and I would prefer to use it here.

Reply:
Done.